### DISCUSSION TITLE
Python support

### DISCUSSION THREAD

**Discussion by Raven888888** [TimefoldAI/timefold-solver#225] [Ideas]
I have been using OptaPlanner and just recently stumble across TimeFold, I am really excited to switch over.
However, all my current codebase is in Python and I am using OptaPy all the time.

TimeFold does mention Python support on its website, and I could find [this pip package](https://pypi.org/project/timefold/), but it seems quite outdated and the version does not indicate a clear relationship with Java TimeFold version.

This makes me **a big hurdle** to migrate over.

Furthermore, the [current quickstart](https://github.com/TimefoldAI/timefold-quickstarts) only covers Java code. Also need a python quickstart, just like [OptaPy](https://github.com/optapy/optapy-quickstarts)

Thanks!

**Comment by triceo** [comment]
Thank you for your interest in Timefold.
At the moment, we do not offer a Python solver. 
That may change in the future, but for the time being, Java/Kotlin are the only languages supported.

Follow us on social media and you'll be among the first to know if/when we introduce the Python solver.

**Comment by Raven888888** [comment]
Thanks @triceo 

Perhaps you should update the [website ](https://timefold.ai/)

![image](https://github.com/TimefoldAI/timefold-solver/assets/58241952/fe9d4842-9976-451e-87f4-d5aea34bc67d)

**Comment by DASHVANTH-03** [comment]
Hey!!
I'm a python beginner......
**Can u help me to write a program to find group of an atom in periodic table.** 
THANK YOU ; D

**Comment by metaleap** [comment]
Python is still prominently listed among Kotlin, Docker etc. on the homepage, yet the search in Docs for Python yields 0 results. What's the story exactly?

**Reply by ge0ffrey** [reply]
We will provide Timefold Solver for Python. More details when we're closer to releasing it.

You'll be able to write a pure Python app, use pip install timefold and solve planning problems with it. Stay tuned...

**Comment by Raven888888** [comment]
Thanks for the update! @ge0ffrey

**Comment by zjevik** [comment]
@ge0ffrey is there any update on python support? I'd be very interested!

**Comment by Christopher-Chianelli** [comment]
@zjevik @Raven888888 [The Python Solver is now released](https://timefold.ai/blog/new-open-source-solver-python).
